app/flatpak-cli-transaction.c

```diff
@@ -1082,6 +1084,46 @@ append_bus (GPtrArray  *talk,
     }
 }
 
+static void
+append_usb (GPtrArray *usb_array,
+            GKeyFile  *metadata,
+            GKeyFile  *old_metadata)
+{
+  gsize i, size = 0;
+  g_auto(GStrv) hidden_devs = NULL;
+  g_auto(GStrv) old_hidden_devs = NULL;
+  g_auto(GStrv) old_enumerables = NULL;
+  g_auto(GStrv) enumerables = g_key_file_get_string_list (metadata, FLATPAK_METADATA_GROUP_USB_DEVICES, FLATPAK_METADATA_KEY_USB_ENUMERABLE_DEVICES,
+                                                          &size, NULL);
+
+  if (old_metadata)
+    old_enumerables = g_key_file_get_string_list (old_metadata, FLATPAK_METADATA_GROUP_USB_DEVICES, FLATPAK_METADATA_KEY_USB_ENUMERABLE_DEVICES,
+                                                  NULL, NULL);
+
+  for (i = 0; i < size; i++)
```

Comment:
Nitpick: something I've been using a lot lately, is for-loops with inlined variable declarations. This makes sure we never use e.g. `i` outside of the `for` block. This seems to be the case here, and declaring variable in the for line is already used within Flatpak codebase, so I think it's worth using it here too.

```suggestion
  for (size_t i = 0; i < size; i++)
```